cpp/log/test.cpp

```diff
@@ -59,6 +64,104 @@ void test() {
 
 	t0.join();
 	t1.join();
+
+	return 0;
+}
+
+
+/**
+ * Creates a FileSink to also write log Messages
+ * into /tmp/openage-log-test.
+ * Only Messages with a level equals or higher
+ * than level::err will be printed in that file.
+ */
+int file_log_sink () {
+	char filelog_name[L_tmpnam];
+	int result = 0;
+
+	tmpnam(filelog_name);
+
+	TestLogSource logger;
+	FileSink filelog(filelog_name, false);
+	filelog.loglevel = log::level::err;
+
+	// filelog should ignore these Messages
+	logger.log(MSG(dbg) << "This message should not appear in openage-log-test");
+	logger.log(MSG(info) << "This message should not appear in openage-log-test");
+	logger.log(MSG(warn) << "This message should not appear in openage-log-test");
+
+	// filelog should write these Messages
+	logger.log(MSG(err) << "This message should appear in openage-log-test");
+	logger.log(MSG(crit) << "This message should appear in openage-log-test");
+
+	std::string line;
+	std::ifstream infile(filelog_name);
+
+	while (std::getline(infile, line)) {
+		if ((line.find("ERR") == std::string::npos) && (line.find("CRIT") == std::string::npos)) {
```

Comment:
i'd recommend searching for `"should not appear"`, because this is the essential part to test, not the line prefix which may change any time.